ui/imports/shared/controls/chat/LinkPreviewSettingsCard.qml

```diff
@@ -0,0 +1,144 @@
+                Layout.fillWidth: true
+                Layout.fillHeight: true
+                font.pixelSize: Style.current.additionalTextSize
+                font.weight: Font.Medium
+                wrapMode: Text.Wrap
+                elide: Text.ElideRight
+                maximumLineCount: 1
+                text: qsTr("Show link previews?")
+            }
+            StatusBaseText {
+                Layout.fillWidth: true
+                Layout.fillHeight: true
+                font.pixelSize: Style.current.additionalTextSize
+                color: Theme.palette.baseColor1
+                wrapMode: Text.Wrap
+                elide: Text.ElideRight
+                maximumLineCount: 1
+                text: qsTr("A preview of your link will be shown here before you send it")
+            }
+        }
+        ComboBox {
+            id: optionsComboBox
+            Layout.leftMargin: 12
+            Layout.preferredHeight: 38
+            leftPadding: 12
+            rightPadding: 12
+            hoverEnabled: true
+            flat: true
+            contentItem: RowLayout {
+                spacing: Style.current.halfPadding
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                    verticalAlignment: Text.AlignVCenter
+                    font.pixelSize: Style.current.additionalTextSize
+                    elide: Text.ElideRight
+                    text: qsTr("Options")
+                    color: Theme.palette.baseColor1
+                }
+                StatusIcon {
+                    Layout.preferredWidth: 16
+                    Layout.preferredHeight: 16
+                    icon: "chevron-down"
+                    color: Theme.palette.baseColor1
+                }
+            }
+            background: Rectangle {
+                border.width: 1
+                border.color: Theme.palette.directColor7
+                color: optionsComboBox.popup.visible ? Theme.palette.baseColor2 : "transparent"
+                radius: Style.current.radius
+                HoverHandler {
+                    cursorShape: Qt.PointingHandCursor
+                    enabled: optionsComboBox.enabled
+                }
+            }
+            popup: ContextMenu {
+                y: - (height + 4)
+                onEnableLinkPreviewForThisMessage: root.enableLinkPreviewForThisMessage()
+                onEnableLinkPreview: root.enableLinkPreview()
+                onDisableLinkPreview: root.disableLinkPreview()
+            }
+            indicator: null
+        }
+
+        StatusFlatRoundButton {
+            id: closeButton
+            Layout.preferredHeight: 38
+            Layout.preferredWidth: 38
+            type: StatusFlatRoundButton.Type.Secondary
+            icon.name: "close"
+            icon.color: Theme.palette.directColor1
+            onClicked: root.dismiss()
+        }
+    }
+
+    component ContextMenu: StatusMenu {
```

Comment:
When there is only one usage of that component, is it easier to keep it inline? The signals could be emitted directly resulting with simpler and shorter code.

Reply:
It's reused for the other cards and this is why I'm exposing it. The same settings menu will be opened when right clicking a link preview.